src/main/java/it/smartcommunitylab/aac/saml/auth/SamlWebSsoAuthenticationRequestFilter.java

```diff
@@ -132,23 +130,22 @@ protected void doFilterInternal(HttpServletRequest request, HttpServletResponse
         }
 
         RelyingPartyRegistration relyingParty = context.getRelyingPartyRegistration();
-        AbstractSaml2AuthenticationRequest samlRequest;
         if (relyingParty.getAssertingPartyDetails().getSingleSignOnServiceBinding() == Saml2MessageBinding.REDIRECT) {
-            samlRequest = sendRedirect(response, context);
+            sendRedirect(response, context, ctx);
         } else {
-            samlRequest = sendPost(response, context);
+            sendPost(response, context, ctx);
         }
-        // persist (abstract) authentication request as per required by OpenSaml4AuthenticationProvider
-        ctx.setSamlAuthenticationRequest(samlRequest);
         logger.debug("sent request");
     }
 
-    private AbstractSaml2AuthenticationRequest sendRedirect(
+    private void sendRedirect(
         HttpServletResponse response,
-        Saml2AuthenticationRequestContext context
+        Saml2AuthenticationRequestContext context,
+        SerializableSaml2AuthenticationRequestContext serializedContext
     ) throws IOException {
         Saml2RedirectAuthenticationRequest authenticationRequest =
             this.authenticationRequestFactory.createRedirectAuthenticationRequest(context);
+        serializedContext.setSamlAuthenticationRequest(authenticationRequest);
```

Comment:
you should not modify the context *after* it has been persisted. It may work, but only because the current implementation keeps an object in memory. If written (for example serialized in a db), you set action will not be reflected in the persisted content.